### ISSUE TITLE
the criteria to accept the move

### ISSUE THREAD

**Issue by YanLiang1102** [oudalab/co-incidents#4]
so based on our design say group a now has event 1, 2, 3 and group b has event 4,5 
so the current score will be , say a( denote affinity value), r (denote repulsion value)

so the current score for this setup will be a12+a13+a23+a45+r14+r15+r24+r25+r34+r35
and ok if we move 1 to the group b 
the value we get will be 
a23+a14+a15+a45+r12+r13+r24+r25+r34+r35

now for simplicity first design we just assume thet aij=-rij
so we get that in order to compare the score we just need to compare that if 
a12+a13-a14-a15>=a14+a15-a12-a13,
so that is just equivalenet to check if a12+a13-a14-a15<0 if this is <0, then we should make the move,
otherwise we should not make the move,
also need to note that the design I have in the current commit, if you doing the simulation a lot of times,it will converge into 1 cluster!!! which is not correct!
(the currrent criteria is before moving the sentene intio an incidence, need to calculate the avarage similarity of this tobe moved sentence with all the sentence already in the incidence, if this value is bigger than some threshhold then we should move it)

**Comment by YanLiang1102** [comment]
This idea still have the issue taht when some cluster get too big, then if run long enough all the stuff will eventually all clustered together so I have 1000 sentences ti simulate,and I run the move for 100000 times, it will eventually has 9 clusters left which only starts with 1000.

**Comment by YanLiang1102** [comment]
1. two things here, first is how to make the group feature of sentences which will stop everything clustering together
2. need to make the code when moving one sentence need to move the whole subincidence with it
3. instead of doing the reclustering of the subincidence each time (when some sentences link to the incidence, we need more clever move).

**Comment by cegme** [comment]
If they are all merging to one cluster there is something wrong with the weights. It helps to have negative features to restrict this.

**Comment by YanLiang1102** [comment]
@cegme  yes, I know exactly why it clustering into one clustring :P, need better features now it always towards the bigger cluster.

**Comment by YanLiang1102** [comment]
Ok @elenamontes  
things we just talked about 

- sample rank weights on each sentence level
-sample rank weights on each subincidence level
-sample rank weights on each incidence level

------for each level we can learn different weights and also  not need to update the feature of the weight if we just see one configuration got accepted with the fearture turned on,
we need to keep a list of say config1, config2, config10 all get accepted and they all have feature
map [0,0,1,1,0] (supporse we  have 5 features, and feature 3 and 4 get turned on by 3 configurations) then we decide to update the weights on feature 3 and 4.


1.
